RecoEgamma/PhotonIdentification/src/PhotonMVABasedHaloTagger.cc

```diff
@@ -0,0 +1,425 @@
+}
+
+void PhotonMVABasedHaloTagger::calmatchedHBHECoordForBothHypothesis(const CaloGeometry* geo,
+                                                                    const reco::Photon* pho,
+                                                                    const HBHERecHitCollection& HBHERecHits) {
+  hcalClusX_samedPhi_ = 0;
+  hcalClusY_samedPhi_ = 0;
+  hcalClusZ_samedPhi_ = 0;
+
+  hcalClusNhits_samedPhi_ = 0;
+  hcalClusE_samedPhi_ = 0;
+
+  hcalClusX_samedR_ = 0;
+  hcalClusY_samedR_ = 0;
+  hcalClusZ_samedR_ = 0;
+  hcalClusNhits_samedR_ = 0;
+  hcalClusE_samedR_ = 0;
+
+  double phoSCEta = pho->superCluster()->eta();
+  double phoSCPhi = pho->superCluster()->phi();
+
+  // Loop over HBHERecHit's
+  for (const auto& hbherechit : HBHERecHits) {
+    HcalDetId det = hbherechit.id();
+    const GlobalPoint& rechitPoint = geo->getPosition(det);
+
+    double rhEta = rechitPoint.eta();
+    double rhPhi = rechitPoint.phi();
+    double rhX = rechitPoint.x();
+    double rhY = rechitPoint.y();
+    double rhZ = rechitPoint.z();
+    double rhE = hbherechit.energy();
+
+    int depth = det.depth();
+
+    if ((det.subdet() == HcalBarrel and (depth < 1 or depth > int(recHitEThresholdHB_.size()))) or
+        (det.subdet() == HcalEndcap and (depth < 1 or depth > int(recHitEThresholdHE_.size())))) {
+      edm::LogWarning("PhotonMVABasedHaloTagger")
+          << " hit in subdet " << det.subdet() << " has an unaccounted for depth of " << depth
+          << "!! Leaving this hit!!";
+      continue;
+    }
+
+    const bool goodHBe = det.subdet() == HcalBarrel and rhE > recHitEThresholdHB_[depth - 1];
+    const bool goodHEe = det.subdet() == HcalEndcap and rhE > recHitEThresholdHE_[depth - 1];
+    if (!(goodHBe or goodHEe))
+      continue;
+
+    if (phoSCEta * rhEta < 0)
+      continue;  ///Should be on the same side of Z
+
+    double rho2 = pow(rhX, 2) + pow(rhY, 2);
+    double dPhi = deltaPhi(phoSCPhi, rhPhi);
+
+    bool isRHBehindECAL = false;
+
+    double dRho2 = pow(rhX - ecalClusX_, 2) + pow(rhY - ecalClusY_, 2);
+
+    ///only valid for the EE; this is 26 cm; hit within 3x3 of HCAL centered at the EECAL xtal
+    if (rho2 >= rho2Min_ECALpos_ && rho2 <= rho2Max_ECALpos_ && dRho2 <= dRho2Max_HCALClus_SamePhi_ &&
+        std::abs(dPhi) < dPhiMax_HCALClus_SamePhi_) {
+      hcalClusX_samedPhi_ += rhX * rhE;
+      hcalClusY_samedPhi_ += rhY * rhE;
+      hcalClusZ_samedPhi_ += rhZ * rhE;
+      hcalClusE_samedPhi_ += rhE;
+      hcalClusNhits_samedPhi_++;
+      isRHBehindECAL = true;
+
+    }  //if(rho>=31 && rho<=172)
+
+    double dR2 = reco::deltaR2(phoSCEta, phoSCPhi, rhEta, rhPhi);
+
+    if (dR2 < dR2Max_HCALClus_SamePhi_ &&
+        !isRHBehindECAL) {  ///dont use hits which are just behind the ECAL in the same phi region
+      hcalClusX_samedR_ += rhX * rhE;
+      hcalClusY_samedR_ += rhY * rhE;
+      hcalClusZ_samedR_ += rhZ * rhE;
+      hcalClusE_samedR_ += rhE;
+      hcalClusNhits_samedR_++;
+    }
```

Comment:
This could allow saving a few cpu cycles:
```suggestion
    if ( !isRHBehindECAL) {
       double dR2 = reco::deltaR2(phoSCEta, phoSCPhi, rhEta, rhPhi);
      ///dont use hits which are just behind the ECAL in the same phi region
       if (dR2 < dR2Max_HCALClus_SamePhi_) {
         hcalClusX_samedR_ += rhX * rhE;
         hcalClusY_samedR_ += rhY * rhE;
         hcalClusZ_samedR_ += rhZ * rhE;
         hcalClusE_samedR_ += rhE;
         hcalClusNhits_samedR_++;
      }
    }
```